src/hostapi/pulseaudio/pa_linux_pulseaudio_cb.c

```diff
@@ -105,6 +105,34 @@ int PaPulseAudio_updateTimeInfo( pa_stream * s,
     return 0;
 }
 
+/* Release pa_operation always same way */
+int PaPulseAudio_ReleaseOperation(PaPulseAudio_HostApiRepresentation *hostapi,
+                                  pa_operation **operation)
+{
+    unsigned int wait = 0;
+    pa_operation *localOperation = (*operation);
+
+    while( pa_operation_get_state( localOperation ) == PA_OPERATION_RUNNING )
+    {
+        pa_threaded_mainloop_wait( hostapi->mainloop );
```

Comment:
According to:
https://freedesktop.org/software/pulseaudio/doxygen/thread-mainloop_8h.html#ac96d19260567d0406350f1b3a14c4bed

the call to pa_threaded_mainloop_wait() needs to be under a lock.

Reply:
Ok made needed changes to have locking when using wait

Reply:
I cannot find the commits where the locks were added.

Reply:
My bad! Not pushed stuff as you mentioned. Now done it and rebased.